ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/ChainDB/API.hs

```diff
@@ -440,6 +438,9 @@ addBlockWaitWrittenToDisk chainDB punish blk = do
 -- block died, in that case 'FailedToAddBlock' will be returned.
 --
 -- Note: this is a partial function, only to support tests.
+--
+-- PRECONDITON: the block to be added must not be from the future. See 'addBlockAsync'.
```

Comment:
You say "See `addBlockAsync`.", but there are actually two `addBlockAsync`s:

 - One is the API function in the record above, I guess that's what this should link to? It currently *doesn't* mention this precondition, maybe it should.
 - The other is the implementation of that API function in `ChainSel.hs`. It does mention the precondition.

Also:
```suggestion
-- PRECONDITION: the block to be added must not be from the future. See 'addBlockAsync'.
```